Views/SensimityView.m

```diff
@@ -0,0 +1,114 @@
+    _running = FALSE;
+    
+    // Add a listener to hide the keyboard if user clickes outside the keyboard
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc]
+                                   initWithTarget:self
+                                   action:@selector(dismissKeyboard)];
+    
+    [self addGestureRecognizer:tap];
+    
+    [_uuid setDelegate:self];
+    [_majorNumber setDelegate:self];
+    [_minorNumber setDelegate:self];
+}
+
+// Start or stop the beaconadvertisement
+- (IBAction)buttonClicked:(id)sender {
+    if (_running) {
+        [self.peripheralManager stopAdvertising];
+        [_toggleButton setTitle:@"START" forState:UIControlStateNormal];
+        [_toggleButton setTitleColor:[UIColor colorWithRed:0.325 green:0.337 blue:0.353 alpha:1] forState:UIControlStateNormal];
+        [_toggleButton setBackgroundColor:[UIColor colorWithRed:0.949 green:0.663 blue:0 alpha:1]];
+        _running = FALSE;
+    } else {
+        // Create a NSUUID object
+        NSUUID *uuid = [[NSUUID alloc] initWithUUIDString:[_uuid text]];
+        
+        // Initialize the Beacon Region
+        CLBeaconRegion *myBeaconRegion = [[CLBeaconRegion alloc] initWithProximityUUID:uuid
+                                                                                 major:[[_majorNumber text] intValue]
+                                                                                 minor:[[_minorNumber text] intValue]
+                                                                            identifier:@"com.enrise.sensimity.chameleon"];
+        // Get the beacon data to advertise
+        self.myBeaconData = [myBeaconRegion peripheralDataWithMeasuredPower:nil];
+        
+        // Start the peripheral manager
+        self.peripheralManager = [[CBPeripheralManager alloc] initWithDelegate:self
+                                                                         queue:nil
+                                                                       options:nil];
+        [_toggleButton setTitle:@"STOP" forState:UIControlStateNormal];
+        [_toggleButton setTitleColor:[UIColor colorWithRed:0.949 green:0.663 blue:0 alpha:1] forState:UIControlStateNormal];
+        [_toggleButton setBackgroundColor:[UIColor colorWithRed:0.325 green:0.337 blue:0.353 alpha:1]];
+        _running = TRUE;
+    }
+}
+
+// Handle start or stop advertising
+-(void)peripheralManagerDidUpdateState:(CBPeripheralManager*)peripheral
+{
+    if (peripheral.state == CBPeripheralManagerStatePoweredOn)
+    {
+        // Bluetooth is on
+        // Start broadcasting
+        [self.peripheralManager startAdvertising:self.myBeaconData];
+    }
+    else if (peripheral.state == CBPeripheralManagerStatePoweredOff)
+    {
+        
+        // Bluetooth isn't on. Stop broadcasting
+        [self.peripheralManager stopAdvertising];
+    }
+    else if (peripheral.state == CBPeripheralManagerStateUnsupported)
+    {
+        // Unsupported
+    }
+}
```

Comment:
If we are planning to put this app in appstore, we should add error and success messages. I made a small search and in Android 'Toasts' are used to display error messages. We need a good design to display error messages. It may be wise to ask Martijn's help in this issue :P